File: playpen/swigtest/applications/testswig.py
#!/usr/bin/env python
#
# ----------------------------------------------------------------------
#
#                           Brad T. Aagaard
#                        U.S. Geological Survey
#
# <LicenseText>
#
# ----------------------------------------------------------------------
#

from pyre.applications.Script import Script as Application

# TestApp class
class TestApp(Application):
  """
  Python TestApp application.
  """
  
  # INVENTORY //////////////////////////////////////////////////////////

  class Inventory(Application.Inventory):
    """
    Python object for managing PyLithApp facilities and properties.
    """

    import pyre.inventory

    from swigtest.Scene import Scene
    scene = pyre.inventory.facility("scene", family="scene", factory=Scene)
    scene.meta['tip'] = "Scene."


  # PUBLIC METHODS /////////////////////////////////////////////////////

  def __init__(self, name="testapp"):
    """
    Constructor.
    """
    Application.__init__(self, name)
    return


  def main(self, *args, **kwds):
    """
    Run the application.
    """
    self.scene.view()

    import numpy
    x = numpy.array([1.1, 2.2, 3.3, 4.4], dtype=numpy.float32)
    self.scene.printData(x)
    return
  

  # PRIVATE METHODS ////////////////////////////////////////////////////

  def _configure(self):
    """
    Setup members using inventory.
    """
    Application._configure(self)
    self.scene = self.inventory.scene
    return

# ======================================================================
if __name__ == '__main__':
  app = TestApp()
  app.run()


# End of file 

File: playpen/swigtest/applications/testswig.cc
// -*- C++ -*-
//
// ======================================================================
//
//                           Brad T. Aagaard
//                        U.S. Geological Survey
//
// {LicenseText}
//
// ======================================================================
//


#include "Bar.hh"
#include "Sphere.hh"
#include "Scene.hh"

int
main(int argc,
     char** argv)
{
  geometry::Bar bar;
  bar.length(4.0);
  bar.width(2.0);
  bar.color("yellow");
  bar.id(0);

  geometry::Sphere sphere;
  sphere.radius(3.0);
  sphere.color("green");
  sphere.id(1);

  geometry::Scene scene;
  scene.bar(&bar);
  scene.sphere(&sphere);
  scene.view();

  return 0;
}


// End of file
